File: TravellerCreatorModels/RTTWorldgen/RTTWorldgenStarSystem.cs
using TravellerCreatorModels.Enums;
using TravellerCreatorModels.Interfaces;

namespace TravellerCreatorModels.RTTWorldgen;

public class RTTWorldgenStarSystem : IStarSystem
{
    public List<IStar> Stars { get; set; } = new();
    public StarSystemType Type { get; set; }
    public bool GasGiant { get; set; }
    public List<IPlanet> Planets { get; set; } = new();
}

File: TravellerCreator/ClientApp/src/app/sector-creator/models/rtt-worldgen/rtt-worldgen.star-system.ts
import {IStarSystem} from "../interfaces/star-system";
import {StarSystemType} from "../enums/star-system-type";
import {RttWorldgenPlanet} from "./rtt-worldgen.planet";
import {RttWorldgenStar} from "./rtt-worldgen.star";

export class RttWorldgenStarSystem implements IStarSystem{
  stars: RttWorldgenStar[];
  type: StarSystemType;

  gasGiant: boolean;
  planets: RttWorldgenPlanet[];
}
